libkbfs/disk_block_cache.go

```diff
@@ -0,0 +1,316 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/kbfscrypto"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/storage"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"golang.org/x/net/context"
+)
+
+const (
+	// 10 GB maximum storage by default
+	defaultDiskBlockCacheMaxBytes uint64 = 10 * (1 << 30)
+	evictionConsiderationFactor   uint   = 3
+	blockDbFilename               string = "diskCacheBlocks.leveldb"
+	lruDbFilename                 string = "diskCacheLRU.leveldb"
+)
+
+// diskBlockCacheEntry packages an encoded block and serverHalf into one data
+// structure, allowing us to encode it as one set of bytes.
+type diskBlockCacheEntry struct {
+	Buf        []byte
+	ServerHalf kbfscrypto.BlockCryptKeyServerHalf
+}
+
+// diskBlockCacheConfig specifies the interfaces that a DiskBlockCacheStandard
+// needs to perform its functions. This adheres to the standard libkbfs Config
+// API.
+type diskBlockCacheConfig interface {
```

Comment:
I disagree. For one, we have default implementations of these interfaces for tests. Also, this is the most flexible way to allow extension in the future (like with strib's suggested use `config.Clock()`).